freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserService.java

```diff
@@ -399,4 +400,18 @@ void validateCrnFilter(Set<String> crnFilter, Crn.ResourceType resourceType) {
     private Set<String> union(Collection<String> collection1, Collection<String> collection2) {
         return Stream.concat(collection1.stream(), collection2.stream()).collect(Collectors.toSet());
     }
+
+    public void modifyAdminPasswordExpirationIfNeeded(FreeIpaClient client) throws FreeIpaClientException {
```

Comment:
The UserService currently only performs the  user sync. I don't think this method really fits in well in the UserService. Consider moving it to another (perhaps new) class.

Reply:
moved it, but I think then `UserService` name is not correct for this class, it should be `UserSyncService`